src/components/SelectionList/types.ts

```diff
@@ -34,25 +35,25 @@ type CommonListItemProps<TItem> = {
 
 type User = {
     /** Text to display */
-    text: string;
+    text?: string;
 
     /** Alternate text to display */
-    alternateText?: string;
+    alternateText?: string | null;
 
     /** Key used internally by React */
-    keyForList: string;
+    keyForList?: string | null;
 
     /** Whether this option is selected */
     isSelected?: boolean;
 
     /** Whether this option is disabled for selection */
-    isDisabled?: boolean;
+    isDisabled?: boolean | null;
```

Comment:
Well, if `null` is really needed in `ReportUtils.OptionData` we should support, otherwise let's delete it so we don't need to cascade this `?: boolean | null;` up to this component! cc @ruben-rebelo

Reply:
I agree, I've reverted all the changes made on SelectionList and updated SearchPage.
Please have a look @ntdiary

Reply:
I'm afraid with this changed we can't avoid the assertion usage.
Do you prefer to revert this one and keep the nulls for this PR?
I'll update the offlineMessage and textInputHint with these suggestions.